libcore/BookmarkList.vala

```diff
@@ -196,25 +201,29 @@ namespace Marlin {
             return list.nth_data (index);
         }
 
-        public void move_item (uint index, uint destination) {
-            if (index > list.length ()) { // Can be assumed to be limited in length
-                critical ("Bookmarklist: Attempt to move bookmark from out of range index");
-                return;
-            }
+        public void move_item_uri (string uri, int step) {
+            bool list_changed = false;
+            unowned GLib.List<Marlin.Bookmark> node = list;
+            unowned GLib.List<Marlin.Bookmark> next = node.next;
 
-            if (destination > list.length ()) { // Can be assumed to be limited in length
-                critical ("Bookmarklist: Attempt to move bookmark to out of range index");
-                return;
-            }
+            int index = 0;
+            for (node = list; node != null; node = next) {
```

Comment:
```suggestion
            foreach (unowned var bm in list) {
```

Reply:
Done.  This was a remnant of the old direct port from C